include/circt/Dialect/Arc/ArcOps.td

```diff
@@ -566,6 +566,68 @@ def StateWriteOp : ArcOp<"state_write", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Simulation Orchestration
+//===----------------------------------------------------------------------===//
+
+def SimInstantiate : ArcOp<"sim.instantiate"> {
```

Comment:
Should this op be `Pure`? As far as I can tell it has no observable side effects.

Reply:
Maybe use an allocate side-effect?